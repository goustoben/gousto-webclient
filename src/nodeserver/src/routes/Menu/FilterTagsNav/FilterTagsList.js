import React, { PropTypes } from 'react'
import classnames from 'classnames'
import css from './FilterTagsList.css'
import FilterTag from './FilterTag'

const FilterTagList = ({ tags }) => (
	<div className={classnames(css.filterTagList, { [css.divider]: tags.length > 0 })}>
		{tags.map(tag => <FilterTag>{tag}</FilterTag>)}
	</div>
)

FilterTagList.propTypes = {
	tags: PropTypes.arrayOf(PropTypes.string),
}

FilterTagList.defaultProps = {
	tags: [],
}

export default FilterTagList
