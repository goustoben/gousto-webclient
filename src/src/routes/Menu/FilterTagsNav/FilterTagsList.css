
.filterTagList {
    width: 100%;
    padding-left: 10px;
    display: flex;
}
