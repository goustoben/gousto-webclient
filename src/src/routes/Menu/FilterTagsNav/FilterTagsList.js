import React, { PropTypes } from 'react'
import classnames from 'classnames'
import css from './FilterTagsList.css'
import FilterTag from '../FilterTag/FilterTagContainer'

const FilterTagList = ({ tags }) => (
	<div className={classnames(css.filterTagList, { [css.divider]: tags.length > 0 })}>
		{tags.map(({ text, type, value }) => <FilterTag key={text} text={text} type={type} value={value} />)}
	</div>
)

FilterTagList.propTypes = {
	tags: PropTypes.arrayOf(
		PropTypes.shape({
			text: PropTypes.string,
			type: PropTypes.string,
			value: PropTypes.string,
		}),
	),
}

FilterTagList.defaultProps = {
	tags: [],
}

export default FilterTagList
