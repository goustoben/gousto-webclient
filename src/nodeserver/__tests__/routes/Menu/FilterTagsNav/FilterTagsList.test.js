import React from 'react';
import renderer from 'react-test-renderer';
import FilterTagsList from 'routes/Menu/FilterTagsNav/FilterTagsList'

describe("FilterTagsList", () => {

  describe("rendering", () => {

    test("Will render FilterTagsList component", () => {
      const component = <FilterTagsList tags={["Gluten Free", "Reactive"]}/>

      const tree = renderer
      .create(component)
      .toJSON()

      expect(tree).toMatchSnapshot()
    })
  })
})
