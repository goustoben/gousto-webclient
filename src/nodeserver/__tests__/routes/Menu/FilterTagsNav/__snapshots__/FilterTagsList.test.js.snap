// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterTagsList rendering Will render FilterTagsList component 1`] = `
<div
  className="filterTagList divider"
>
  <div
    className="filterTag"
  >
    <span
      className="fa-times"
      style={
        Object {
          "color": "#939596",
          "fontSize": "inherit",
        }
      }
    />
    Gluten Free
  </div>
  <div
    className="filterTag"
  >
    <span
      className="fa-times"
      style={
        Object {
          "color": "#939596",
          "fontSize": "inherit",
        }
      }
    />
    Reactive
  </div>
</div>
`;
